ts/packages/cli/src/helpers/stores/index.ts

```diff
@@ -1,26 +1,24 @@
-import { URL } from 'url';
-import { ProtocolToInit } from './types';
-
+import { URI } from '@configu/ts';
+import { SchemeToInit } from './types';
 import { NoopStorePTI } from './Noop';
-import { ConfiguStorePTI } from './Configu';
-import { JsonFileStorePTI } from './JsonFile';
-import { HashiCorpVaultStorePTI } from './HashiCorpVault';
-import { AwsSecretsManagerStorePTI } from './AwsSecretsManager';
+import { ConfiguStoreSTI } from './Configu';
+import { JsonFileStoreSTI } from './JsonFile';
+import { HashiCorpVaultStoreSTI } from './HashiCorpVault';
+import { AwsSecretsManagerStoreSTI } from './AwsSecretsManager';
 
-const PROTOCOL_TO_STORE_INIT_FN_DICT: ProtocolToInit = {
+const SCHEME_TO_STORE_INIT_FN_DICT: SchemeToInit = {
   ...NoopStorePTI,
-  ...ConfiguStorePTI,
-  ...JsonFileStorePTI,
-  ...HashiCorpVaultStorePTI,
-  ...AwsSecretsManagerStorePTI,
+  ...ConfiguStoreSTI,
+  ...JsonFileStoreSTI,
+  ...HashiCorpVaultStoreSTI,
+  ...AwsSecretsManagerStoreSTI,
 };
 
-export const constructStoreFromUrl = (url: string) => {
-  const storeUrl = new URL(url);
-  const storeProtocol = storeUrl.protocol.slice(0, -1);
-  const storeInitFunction = PROTOCOL_TO_STORE_INIT_FN_DICT[storeProtocol];
+export const constructStoreFromUri = (uri: string) => {
```

Comment:
@rannn505 

The change we made causes us to duplicate the following lines for all stores:
``` node.js
    const parsedUri = URI.parse(uri);
    const queryDict = _.fromPairs(parsedUri.query?.split('&').map((query) => query.split('=')));
    const splittedUserinfo = parsedUri.userinfo?.split(':');
```

I suggest that we adjust the `SchemeToInit` logic to supply these details directly to each store to prevent duplication. If we do so then we need to adjust the @configu/ts to export the `URIComponents` type. Should we do it now or open an issue for this for later?

Reply:
@rannn505 & I decided to tackle this now.